cmd/fs-v1-multipart.go

```diff
@@ -678,12 +680,9 @@ func (fs fsObjects) CompleteMultipartUpload(bucket string, object string, upload
 
 	fsAppendMeta, err := readFSMetadata(fs.storage, minioMetaBucket, fsAppendMetaPath)
 	if err == nil && isPartsSame(fsAppendMeta.Parts, parts) {
-		fsAppendDataPath := getFSAppendDataPath(uploadID)
 		if err = fs.storage.RenameFile(minioMetaBucket, fsAppendDataPath, bucket, object); err != nil {
 			return "", toObjectErr(traceError(err), minioMetaBucket, fsAppendDataPath)
 		}
-		// Remove the append-file metadata file in tmp location as we no longer need it.
-		fs.storage.DeleteFile(minioMetaBucket, fsAppendMetaPath)
```

Comment:
you can keep this line and move the below new two lines you have added into else block. That way it will be more clear.